### PR TITLE
align search form and results page with designs

### LINKED ISSUE #2589: Make search result sidenav better
Search result side nav doesn't fully align with designs / could do with better autopopulation of the anchor links. Right now, https://github.com/CDCgov/phdi/blob/76e39e7c410d7ad9ee6c728450e848d26961eb0f/containers/tefca-viewer/src/app/query/components/SideNav.tsx#L99 to make nav items with, which makes for some potentially unwanted items when the selector is wrong. 

## Implementation
(note the inclusion of the "How is my data stored?" that is being grabbed accidentally)

### PR DESCRIPTION
**Pull request by fzhao99**
# PULL REQUEST

## Summary
Various styling tweaks, frontend tech debt refactors, and improvements to align things with Figma that Bob worked on when everyone else was out of office last week. Should result in the results page and the search form being up to date / ahead of design minus unimplemented features (step indicator, search bar on results view). 

Of note:

- _**WOULD LOVE OTHER DEVS' OPINIONS ON SCAFFOLDING OUT THIS FOLDER STRUCTURE**_. The creation of separate `designSystem` directory for any context-unaware components (the accordion, redirect toast, side nav (newly introduced in this PR) , and site alert components for now). Breaking these out to implement some [Atomic Design](https://atomicdesign.bradfrost.com/chapter-2/) ideas and to eventually make building something like [Storybook](https://storybook.js.org/) or [Figma Code Connect](https://github.com/figma/code-connect) easier.
     - If I could have my way, we'd also have a separate folder for the page / view level, so that we'd have 
         - `designSystem` for any components that come out of the Figma design system
         - `components` for things in the middle and 
         - `pages` that match the routing / actual pages that map into the URL's people visit. 
         - If we do go down this route, this arch should start to reflect larger design patterns around pushing state away from 1) UI implementation at the `designSystem` level and 2) page routing at the `pages` level. This will be explored further in a writeup that I'll work on related to work called out in #2573. 
- Modularizing a lot of our CSS / putting components into separate folders rather than just living in `components`.
- Seeding out some spacing tokens (specifically `$tefca-viewer-smaller-width`) that we'll probably want to systematize / dedupe from USWDS settings, which I made a ticket for [here](https://app.zenhub.com/workspaces/skylight-dibbs-viper-6480ba23b5a07644e0e46d23/issues/gh/cdcgov/phdi/2611). 

### [Search form](https://www.figma.com/design/YU5hAHY1Eg7J6gtNbBxm7d?node-id=3576-42191&m=dev#957429803)
Laid out the dropdowns horizontally rather than vertically and added explainer copy within the query selection body.
Screenshot 2024-09-23 at 3 52 29 PM

### [Results view](https://www.figma.com/design/YU5hAHY1Eg7J6gtNbBxm7d/DIBBs-Site-Experiences?node-id=3576-42017&m=dev)
Aligned the side nav styling and the backlink/new patient search button. Right now, these link back to the same search form, but when we build out the patient search vs the saved query selection, these will link out to their respective locations. Also adds the name of the query in the title section.
Screenshot 2024-09-23 at 3 52 58 PM

## Related Issue
Fixes #2589 

## Additional Information
Anything else the review team should know?

### CODE REVIEW

File: containers/tefca-viewer/e2e/query_workflow.spec.ts [hunk 1]
```diff
@@ -270,17 +271,17 @@ test.describe("Test the user journey of a 'tester'", () => {
     );
 
     // Check that there are multiple rows in the table
-    await expect(page.locator("tbody").locator("tr")).toHaveCount(9);
+    await expect(page.locator("tbody").locator("tr")).toHaveCount(10);
```
Comment (fzhao99) [inline containers/tefca-viewer/e2e/query_workflow.spec.ts:274]: It seems at some point we merged a data-related PR that changed the outcome of this. Checked this against a cut of main and got the [same result](https://github.com/CDCgov/phdi/actions/runs/11000458933/job/30542906485?pr=2610)

File: containers/tefca-viewer/src/app/query/page.tsx [hunk 2]
```diff
@@ -63,70 +64,74 @@ const Query: React.FC = () => {
...
-            setMode={setMode}
-            setLoading={setLoading}
-            setUseCaseQueryResponse={setUseCaseQueryResponse}
-            setOriginalRequest={setOriginalRequest}
-            setQueryType={setQueryType}
-          />
-        </Suspense>
-      )}
+    <>
+      <SiteAlert page={mode} />
```
Comment (fzhao99) [inline containers/tefca-viewer/src/app/query/page.tsx:68]: The diff here is struggling a bit but the only thing added here was lifting up `<SiteAlert>` up a level and adding `main-container` to the first div

File: containers/tefca-viewer/src/app/query/components/resultsView/ResultsViewAccordionBody.tsx [hunk 3]
```diff
@@ -22,9 +22,11 @@ const ResultsViewAccordionBody: React.FC<ResultsViewAccordionBodyProps> = ({
 }) => {
   return (
     <>
-      <h4 id={id} className={styles.accordionHeading}>
-        {title}
-      </h4>
+      {title && (
```
Comment (fzhao99) [inline containers/tefca-viewer/src/app/query/components/resultsView/ResultsViewAccordionBody.tsx:25]: Adding this check based on a conversation I had with @mikang as documented [here](https://www.figma.com/design/YU5hAHY1Eg7J6gtNbBxm7d?node-id=3576-42017&m=dev#959373362)

**Review by robertandremitchell — APPROVED** [review]
Overall, I really like this format. I'm gonna read through more of the docs on the principles today, but I think it makes a lot of sense and will make managing UI and backend changes a lot more seamless than it has been.

Thanks so much for your work in starting this refactoring.

**Comment by m-goggins** [comment]
I'm taking a look at this PR, just taking my time to read through the docs to make sure I understand the changes you are proposing.

**Pull request merged**